### PR TITLE
Use TRD tracks in TOF matching QC (if requested)

### PR DESCRIPTION
**Pull request by chiarazampolli**
@noferini 
Once this is merged, I can enable the new QC in O2DPG (see https://github.com/AliceO2Group/O2DPG/pull/285).
Missing is also the json for data. I will take care maybe in another PR, but in time for apass4 (even if not really needed since we will most likely exclude TRD).

### CODE REVIEW

File: Modules/TOF/src/TOFMatchedTracks.cxx [hunk 1]
```diff
@@ -337,24 +444,50 @@ void TOFMatchedTracks::monitorData(o2::framework::ProcessingContext& ctx)
...
       LOG(debug) << "DEN CONSTR: track with eta " << tpcTrack.getEta() << " and pT " << tpcTrack.getPt() << " ACCEPTED for denominator CONSTR, ITSTPC track";
-      this->mInTracksPt[trkType::CONSTR]->Fill(tpcTrack.getPt());
-      this->mInTracksEta[trkType::CONSTR]->Fill(tpcTrack.getEta());
-      this->mInTracks2DPtEta[trkType::CONSTR]->Fill(tpcTrack.getPt(), tpcTrack.getEta());
+      this->mInTracksPt[matchType::ITSTPC_ITSTPCTRD]->Fill(tpcTrack.getPt());
+      this->mInTracksEta[matchType::ITSTPC_ITSTPCTRD]->Fill(tpcTrack.getEta());
+      this->mInTracks2DPtEta[matchType::ITSTPC_ITSTPCTRD]->Fill(tpcTrack.getPt(), tpcTrack.getEta());
     }
+    // In case of ITS-TPC-TRD-TOF, TPC-TRD, ITS-TPC-TRD-TOF, TPC-TRD-TOF, it is always a TRD track
+    if constexpr (isTRDTrack<decltype(trk)>()) {
```
Comment (chiarazampolli) [inline Modules/TOF/src/TOFMatchedTracks.cxx:452]: Hello @shahor02 , 

Can you check that the selections in the lambda function are correct for the various track types? The resulting efficiencies look reasonable to me, but it is not so obvious to use the indices etc with the RecoContainer functionalities.

**Review by njacazio — APPROVED** [review]
Thanks Chiara!

**Comment by chiarazampolli** [comment]
Hello @Barthelemy , @knopers8 , @ktf , @TimoWilken 

Running git clang-format, I get:

```
[O2PDPSuite/latest-o2] ~/SOFT/alibuild/ali-o2-dev/QualityControl $> git clang-format upstream/master
Configuration file(s) do(es) not support Json: /home/zampolli/SOFT/alibuild/ali-o2-dev/QualityControl/.clang-format
error: `clang-format -lines=1:76 Modules/TOF/tofMatchedTracks_ITSTPCTOF_TPCTOF_ITSTPCTRDTOF_direct_MC.json` failed
```

How can I fix it?
Chiara

File: Modules/TOF/CMakeLists.txt [hunk 2]
```diff
@@ -112,6 +112,7 @@ install(FILES tofdigits.json
               tofMatchedTracks_ITSTPCTOF_TPCTOF_MC.json
               tofMatchedTracks_ITSTPCTOF_TPCTOF_direct.json
               tofMatchedTracks_ITSTPCTOF_TPCTOF_direct_MC.json
+	      tofMatchedTracks_ITSTPCTOF_TPCTOF_ITSTPCTRDTOF_direct_MC.json
```
Comment (shahor02) [inline Modules/TOF/CMakeLists.txt:115]: Why there is ITS-TPC-TRD-TOF and  not TPC-TRD-TOF?

File: Modules/TOF/include/TOF/TOFMatchedTracks.h [hunk 3]
```diff
@@ -72,34 +77,41 @@ class TOFMatchedTracks final : public TaskInterface
   std::shared_ptr<o2::globaltracking::DataRequest> mDataRequest;
   o2::globaltracking::RecoContainer mRecoCont;
   GID::mask_t mSrc = GID::getSourcesMask("ITS-TPC");
-  GID::mask_t mAllowedSources = GID::getSourcesMask("TPC,ITS-TPC,TPC-TOF,ITS-TPC-TOF");
+  GID::mask_t mAllowedSources = GID::getSourcesMask("TPC,ITS-TPC,TPC-TOF,ITS-TPC-TOF,TPC-TRD,ITS-TPC-TRD-TOF,ITS-TPC-TRD-TOF");
```
Comment (shahor02) [inline Modules/TOF/include/TOF/TOFMatchedTracks.h:80]: You have twice ITS-TPC-TRD-TOF but not TPC-TRD-TF

File: Modules/TOF/src/TOFMatchedTracks.cxx (hunk 1 above)
Comment (shahor02) [reply Modules/TOF/src/TOFMatchedTracks.cxx:452]: Looks ok

File: Modules/TOF/CMakeLists.txt (hunk 2 above)
Comment (chiarazampolli) [reply Modules/TOF/CMakeLists.txt:115]: A typo while I tried to add TRD back, thanks for spotting (there was also TPC-TRD missing)

File: Modules/TOF/include/TOF/TOFMatchedTracks.h (hunk 3 above)
Comment (chiarazampolli) [reply Modules/TOF/include/TOF/TOFMatchedTracks.h]: A typo while I tried to add TRD back, thanks for spotting (there was also TPC-TRD missing)

**Comment by chiarazampolli** [comment]
https://github.com/AliceO2Group/QualityControl/pull/1103#discussion_r830391399

Thanks for checking, @shahor02 !!

**Comment by Barthelemy** [comment]
> Hello @Barthelemy , @knopers8 , @ktf , @TimoWilken
> 
> Running git clang-format, I get:
> 
> ```
> [O2PDPSuite/latest-o2] ~/SOFT/alibuild/ali-o2-dev/QualityControl $> git clang-format upstream/master
> Configuration file(s) do(es) not support Json: /home/zampolli/SOFT/alibuild/ali-o2-dev/QualityControl/.clang-format
> error: `clang-format -lines=1:76 Modules/TOF/tofMatchedTracks_ITSTPCTOF_TPCTOF_ITSTPCTRDTOF_direct_MC.json` failed
> ```
> 
> How can I fix it? Chiara

Hi, 

To fix it myself, I added 
```
[clangFormat]
    extensions = "cpp,h,hpp,hxx,cxx,c"
``` 
to my `.git/config` in the `QualityControl` folder. 
It avoids checking the format of any other type of files.

**Comment by chiarazampolli** [comment]
Thanks @Barthelemy , it worked!

File: Modules/TOF/CMakeLists.txt [hunk 4]
```diff
@@ -112,6 +112,7 @@ install(FILES tofdigits.json
               tofMatchedTracks_ITSTPCTOF_TPCTOF_MC.json
               tofMatchedTracks_ITSTPCTOF_TPCTOF_direct.json
               tofMatchedTracks_ITSTPCTOF_TPCTOF_direct_MC.json
+	      tofMatchedTracks_ITSTPCTOF_TPCTOF_AllTypes_direct_MC.json
```
Comment (njacazio) [inline Modules/TOF/CMakeLists.txt:115]: Ciao @chiarazampolli could it be that this file was not committed? or that it should be renamed to/from `tofMatchedTracks_AllTypes_direct_MC.json`?

Comment (chiarazampolli) [reply Modules/TOF/CMakeLists.txt:115]: I did fix it a while ago:

```
[O2PDPSuite/latest-o2] ~/SOFT/alibuild/ali-o2-dev/QualityControl $> git diff
diff --git a/Modules/TOF/CMakeLists.txt b/Modules/TOF/CMakeLists.txt
index a8dee66d..343a2e50 100644
--- a/Modules/TOF/CMakeLists.txt
+++ b/Modules/TOF/CMakeLists.txt
@@ -112,7 +112,7 @@ install(FILES tofdigits.json
               tofMatchedTracks_ITSTPCTOF_TPCTOF_MC.json
               tofMatchedTracks_ITSTPCTOF_TPCTOF_direct.json
               tofMatchedTracks_ITSTPCTOF_TPCTOF_direct_MC.json
-             tofMatchedTracks_ITSTPCTOF_TPCTOF_ITSTPCTRDTOF_direct_MC.json
+             tofMatchedTracks_ITSTPCTOF_TPCTOF_AllTypes_direct_MC.json
         DESTINATION etc)
 
 get_property(dirs
[O2PDPSuite/latest-o2] ~/SOFT/alibuild/ali-o2-dev/QualityControl $> git add Modules/TOF/CMakeLists.txt
[O2PDPSuite/latest-o2] ~/SOFT/alibuild/ali-o2-dev/QualityControl $> git commit -m "forgot to change name of file in CMakeLists"
[QCtofTrackTypes 8a9a6f34] forgot to change name of file in CMakeLists
 1 file changed, 1 insertion(+), 1 deletion(-)
[O2PDPSuite/latest-o2] ~/SOFT/alibuild/ali-o2-dev/QualityControl $> git rebase -i upstream/master
Successfully rebased and updated refs/heads/QCtofTrackTypes.
[O2PDPSuite/latest-o2] ~/SOFT/alibuild/ali-o2-dev/QualityControl $> git push -f origin HEAD:updateQCtof
Enumerating objects: 20, done.
Counting objects: 100% (20/20), done.
Delta compression using up to 32 threads
Compressing objects: 100% (11/11), done.
Writing objects: 100% (11/11), 2.15 KiB | 2.15 MiB/s, done.
Total 11 (delta 8), reused 0 (delta 0)
remote: Resolving deltas: 100% (8/8), completed with 8 local objects.
To github.com:chiarazampolli/QualityControl.git
 + 826f03a0...759a0f2a HEAD -> updateQCtof (forced update)
[O2PDPSuite/latest-o2] ~/SOFT/alibuild/ali-o2-dev/QualityControl $> 
```
I have to check why it was not propagated.

Comment (chiarazampolli) [reply Modules/TOF/CMakeLists.txt:115]: It is there:
https://github.com/AliceO2Group/QualityControl/compare/826f03a0e3fc83902605649368a8255176d660a7..759a0f2a05f0a64637d47e4c4132af0f9192371f

Comment (chiarazampolli) [reply Modules/TOF/CMakeLists.txt:115]: Found it, I made a typo.

**Comment by chiarazampolli** [comment]
Hello @Barthelemy , 

Could we merge it? To have it in the nightlies...

Chiara

**Comment by chiarazampolli** [comment]
Thanks @njacazio !

**Comment by Barthelemy** [comment]
@chiarazampolli I did not see and review the PR because the TOF directory is "owned" by Nicolò thus he is the one pinged and assigned.

**Comment by chiarazampolli** [comment]
Hi @Barthelemy , 
Sure, I thought only you and Piotr could merge. Thanks anyway!

**Pull request merged**